### PR TITLE
Updated Caching

### PR DESCRIPTION
**Pull request by elibixby**

### CODE REVIEW

File: bigquery/samples/utils.py [hunk 1]
```diff
@@ -15,8 +15,12 @@
 
 # [START get_service]
 def get_service():
-    from discovery_doc import build_and_update
-    return build_and_update('bigquery', 'v2')
+    from googleapiclient.discovery import build_from_document
+    from oauth2client.client import GoogleCredentials
+    credentials = GoogleCredentials.get_application_default()
+    if credentials.create_scoped_required():
+        credentials = credentials.create_scoped('https://www.googleapis.com/auth/bigquery')
```
Comment (jonparrott) [inline bigquery/samples/utils.py]: Is create_scoped required here?

**Comment by jeffmendoza** [comment]
What is the strategy with the READMEs?

**Comment by elibixby** [comment]
I think we should put information in the highest directory such that it applies to all subdirectories. This way we avoid duplication, which is the purpose of these changes (just moved two identical files into their shared parent).

**Comment by jeffmendoza** [comment]
Cool, sounds good

**Comment by jeffmendoza** [comment]
more Squashage

File: bigquery/samples/utils.py [hunk 2]
```diff
@@ -15,8 +15,12 @@
...
+    return build('bigquery','v2', credentials=GoogleCredentials.get_application_default())
```
Comment (SurferJeffAtGoogle) [inline bigquery/samples/utils.py]: When I run tox, I get this error:
(python-docs-samples)rennie@rennie:~/gitrepos/python-docs-samples$ tox
py27-nosetest installed: Flask==0.10.1,google-api-python-client==1.4.0,httplib2==0.9.1,itsdangerous==0.24,Jinja2==2.7.3,MarkupSafe==0.23,mock==1.0.1,nose==1.3.6,NoseGAE==0.5.7,oauth2client==1.4.11,pyasn1==0.1.7,pyasn1-modules==0.0.5,rsa==3.1.4,simplejson==3.7.1,six==1.9.0,uritemplate==0.6,Werkzeug==0.10.4
py27-nosetest runtests: PYTHONHASHSEED='4074483713'
py27-nosetest runtests: commands[0] | nosetests --with-gae
EEEEE..S.................
======================================================================
ERROR: test_async_query (bigquery.test.test_async_query.TestAsyncQuery)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/google/home/rennie/gitrepos/python-docs-samples/bigquery/test/test_async_query.py", line 28, in test_async_query
    5):
  File "/usr/local/google/home/rennie/gitrepos/python-docs-samples/bigquery/samples/async_query.py", line 48, in run
    service = get_service()
  File "/usr/local/google/home/rennie/gitrepos/python-docs-samples/bigquery/samples/utils.py", line 23, in get_service
    return build('bigquery','v2', credentials=GoogleCredentials.get_application_default())
NameError: global name 'build' is not defined

**Comment by SurferJeffAtGoogle** [comment]
My prior errors were due to the fact that I clicked the "Google internal-only" box when I created the project.

Now I'm seeing this error:
cover runtests: PYTHONHASHSEED='3998625439'
cover runtests: commands[0] | nosetests --with-gae --with-xunit --with-xcoverage --cover-tests --cover-branches --cover-min-percentage=70 --cover-inclusive --cover-erase
...E...S..................
======================================================================
ERROR: test_stream_row_to_bigquery (bigquery.tests.test_streaming.TestStreaming)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/google/home/rennie/gitrepos/python-docs-samples/bigquery/tests/test_streaming.py", line 37, in test_stream_row_to_bigquery
    5):
  File "/usr/local/google/home/rennie/gitrepos/python-docs-samples/bigquery/samples/streaming.py", line 53, in run
    num_retries)
  File "/usr/local/google/home/rennie/gitrepos/python-docs-samples/bigquery/samples/streaming.py", line 40, in stream_row_to_bigquery
    body=insert_all_data).execute(num_retries=num_retries)
  File "/usr/local/google/home/rennie/google-cloud-sdk/platform/google_appengine/lib/oauth2client/oauth2client/util.py", line 129, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/google/home/rennie/gitrepos/python-docs-samples/.tox/cover/local/lib/python2.7/site-packages/googleapiclient/http.py", line 729, in execute
    raise HttpError(resp, content, uri=self.uri)
HttpError: <HttpError 404 when requesting https://www.googleapis.com/bigquery/v2/projects/surferjeff-test2/datasets/test_dataset/tables/test_table2/insertAll?alt=json returned "Not found: Dataset surferjeff-test2:test_dataset">
-------------------- >> begin captured logging << --------------------
googleapiclient.discovery: INFO: URL being requested: GET https://www.googleapis.com/discovery/v1/apis/bigquery/v2/rest
googleapiclient.discovery: INFO: URL being requested: POST https://www.googleapis.com/bigquery/v2/projects/surferjeff-test2/datasets/test_dataset/tables/test_table2/insertAll?alt=json
oauth2client.client: INFO: Attempting refresh to obtain initial access_token
oauth2client.client: INFO: Refreshing access_token
--------------------- >> end captured logging << ---------------------

**Comment by tmatsuo** [comment]
@SurferJeffAtGoogle 
Yep, I also encountered this error.
I found that, in order to use your own project for the test, you need to create dataset and table defined in the constants file.

Ideally, the test setup should be able to take care of them.
At least, we should update the README for how to run the tests.

**Comment by elibixby** [comment]
closed to merge from branch.

**Pull request closed without merging**